### PR TITLE
span: make tests context independent

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
Also make the semi-integration tests real integration tests.

(Part of #1936)

### CODE REVIEW

File: tests/tracer/test_span.py [hunk 1]
```diff
@@ -138,17 +137,11 @@ def __repr__(self):
...
-
+        s = Span(self.tracer, "test.span")
         sleep = 0.05
-        with s as s1:
-            assert s is s1
-            time.sleep(sleep)
+        time.sleep(sleep)
+        s.finish()
         assert s.duration >= sleep, "%s < %s" % (s.duration, sleep)
-        self.assert_span_count(1)
```
Comment (Yun-Kim) [inline tests/tracer/test_span.py:151]: Nitpicking, but can we also check that the span has been recorded? Is there a reason the `self.assert_span_count(1)` was removed?

**Review by Yun-Kim — CHANGES_REQUESTED** [review]
Looks great otherwise 👍

File: tests/tracer/test_span.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply tests/tracer/test_span.py:151]: yes, good catch! the span should still be generated in this case so we should be able to keep the assertion :+1:

Comment (Kyle-Verhoog) [reply tests/tracer/test_span.py:151]: oh, actually on second thought the `Context` is required here 😞 so the span won't be recorded

Comment (Yun-Kim) [reply tests/tracer/test_span.py:151]: Oh okay, makes sense... Can we then update the test's docstring to be a bit more specific?
I.e. `# ensure span.finish() marks the end time of the span and can be recorded`

Comment (Kyle-Verhoog) [reply tests/tracer/test_span.py:151]: sure, we can also remove the `tracer` usage since it's not necessary for this test

Comment (Yun-Kim) [reply tests/tracer/test_span.py:151]: Yeah, I'm good with that as well :)

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**